File: docs/protocol/mqtt/js/qubitro_mqtt_subscribe.js
var mqtt = require('mqtt')

var broker = 'mqtts://broker.qubitro.com'
var deviceId  = 'PASTE_DEVICE_ID'
var deviceToken  = 'PASTE_DEVICE_TOKEN'

var options = {
  keepalive: 60,
  port: 8883,
  username: deviceId,
  password: deviceToken,
  clientId: deviceId,
  clean: false,
  secureProtocol: 'TLSv1_method',
};

var client = mqtt.connect(broker, options);

client.on('connect', function() {
    console.log("Connected to Qubitro!");
    client.subscribe("TOPIC");
    client.on('message', function (topic, message) {
    console.log("Received '" + message + "' on '" + topic + "'");
    });
});

client.on('error', function(err) {
    console.log(err);
});

File: docs/protocol/mqtt/ruby/qubitro_mqtt_subscribe.rb
require 'qubitro-mqtt'

deviceID = "PASTE_DEVICE_ID"
deviceToken = "PASTE_DEVICE_TOKEN"
host = "broker.qubitro.com"

MQTT::Client.connect(
  :host => host,
  :port => 8883,
  :ssl => true,
  :client_id => deviceID,
  :device_id => deviceID,
  :device_token => deviceToken,
  :ack_timeout => 15,
  :will_topic => deviceID,
  ) do |client|
    puts 'Connected to Qubitro!'
    client.get("PASTE_TOPIC") do |topic, msg|
      puts "Received data '#{msg}' on topic '#{topic}'"
    end
end 